apps/microtvm/zephyr/template_project/launch_microtvm_api_server.sh

```diff
@@ -0,0 +1,41 @@
+#!/bin/bash
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+function show_usage() {
+    cat <<EOF
+This script is for running microtvm_api_server.
+Usage: launch_microtvm_api_server.sh <microtvm_api_server.py> --read-fd <READ_FD_PATH> --write-fd <WRITE_FD_PATH>
+EOF
+}
+
+if [ "$#" -lt 5 -o "$1" == "--help" ]; then
+    show_usage
+    exit -1
+fi
+
+west_file_path=$(which west)
+
+# Remove space and extra characters
+line=$(head -n 1 ${west_file_path})
```

Comment:
I don't recall different `west` script version request in its #! a Python binary which includes the minor version, like `#!/usr/bin/python3.8`, which we can't just do `which python3` here instead?

Reply:
what we are trying to do here is to reuse the python that west was installed in and separate that from TVM python ENV. The ideal solution is to create a virtual python ENV for zephyr, install west in it and use that. That seems a bit complicated, so I decided to postpone that for now. Currently, Zephyr is installed in the global python3.8 vs TVM which is in a pythoon3.7 virtual ENV

Reply:
I see.

So, a side effect now is that when `PYTHON_CMD` is set based on `which("python3.8")` if the user has no python3,8 installed in the env command like that use the Project API server, like `tvmc micro create-project`, will quite saying that `Error: Option 'west_cmd' is required but was not specified. Use --list-options to see all required options.`, which is misleading. 

Hence I wonder if the right env isolation should actually happen in the launcher, i.e the launcher would do the check for the right Python version -- 3.8 -- (and later the virtual env), whilst the Project API server would continue to use `sys.executable` to find out the west module.

Reply:
Also, it doesn't seem right to me to check the python used in the `west` shebang and use it to start the Project API server.

Reply:
`sys.executable` doesn't work since it will return tvm python env in docker. I think it should be python3.8 specifically and later we change it to use the python ENV that we use to setup Zephyr/west. Let me know what you think.

Reply:
Yep, I got it. I think that it would be right to adjust the python version to be used in the API server inside the launcher, instead of hardcoding the requested version inside the Project API server. Also this way `sys.executable` would return the right version, no?

Reply:
no, if we you use `sys.executable` it will still pickup the default global Python version. We should change it to ENV later.

Reply:
If you call python3.8 from the launcher to launch the server will the server (i.e. `sys.executable` called from the server) still report the global Python version -- which I'm assuming is Python 3.7?

Reply:
no, in that case it would be python3.8. I think you're right, we can keep using sys.executable since it would follow the Python ENV that called the python script